### PR TITLE
Feat: add Heiko kbtc

### LINKED ISSUE #43: Support Parallel Heiko XCM
Support XCM between Parallel Heiko and Kintsugi for:

- [ ] sKSM
- [ ] kBTC

### PR DESCRIPTION
**Pull request by bvotteler**
Resolves #43 

Add XCM support for Heiko: KBTC

Links to constructed test tx:
- [kintsugi -> heiko](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fapi-kusama.interlay.io%2Fparachain#/extrinsics/decode/0x5e00000b00e40b54020000000000000000000000010102009520010034b83f611d47a5acf4c641b0ee86d746ffca91e1bb616a00a0f43e96184313780008d6e829000000)
- [heiko -> kintsugi](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fheiko-rpc.parallel.fi#/extrinsics/decode/0x2b007900000000e40b5402000000000000000000000001010200b120010034b83f611d47a5acf4c641b0ee86d746ffca91e1bb616a00a0f43e961843137800)

### CODE REVIEW

File: src/adapters/interlay.ts [hunk 1]
```diff
@@ -55,6 +55,20 @@ export const kintsugiRoutersConfig: Omit<CrossChainRouterConfigs, "from">[] = [
...
+    to: "heiko",
+    token: "KBTC",
+    xcm: {
+      // estimated fee calculated from:
+      // ksm_per_sec = ~233_100_000_000
+      // kint_per_sec = (ksm_per_second * 4) / 3 = 310_800_000_000
+      // xcm_secs = ~(247_189_333 / kint_per_sec) ... calculated using a recent kint transaction
+      // kbtc_per_sec = ksm_per_sec / 1_500_000 ... see https://github.com/interlay/interbtc/blob/18efc1e9194a8591713b9885799a6deb6c0392f2/parachain/runtime/kintsugi/src/xcm_config.rs#L104
+      // estimated kbtc fee = kbtc_per_sec * xcm_secs = ~123.6
+      fee: { token: "KBTC", amount: "200" },
```
Comment (bvotteler) [inline src/adapters/interlay.ts:68]: @sander2 , can you please double check my math here?

**Comment by sander2** [comment]
Since we will have a bunch of these prs, I would propose to setup some PR guidelines. Specifically, before merging such prs, I'd like to see some kind of proof that generated extrinsic actually works. This could be: 

- If the channel and both chains are set up, then:
  - if there are recent transactions already, provide a subscan link to a historic successful transfer, plus the polkadotjs link of the generated encoded extrinsic. Reviewers can compare the both.
  - if no transactions were made yet, use the code to make a test tx yourself and provide the subscan link to that tx here. Reviewers can verify that it works
- If the channel or one of the chains is not ready yet, provide only a polkadotjs link of the generated encoded extrinsic. Reviewers can verify that the output looks plausible

**Comment by bvotteler** [comment]
> Since we will have a bunch of these prs, I would propose to setup some PR guidelines. Specifically, before merging such prs, I'd like to see some kind of proof that generated extrinsic actually works. This could be...

Good suggestion, thank you.

For these PRs, I will not have the means to test the transactions (no funds to shuttle across the chains repeatedly). I'm not interested in gaining access to Interlay's test accounts, either. It's more secure the fewer people have access, plus there will be extensive tests once this is added to the UI.

It's enough easy to provide (links to) the generated extrinsics, so providing them is a no-brainer. 
Would it make sense to search for and link similar xcm transactions (same token) to other chains for comparison if they exist?

**Comment by sander2** [comment]
> > Since we will have a bunch of these prs, I would propose to setup some PR guidelines. Specifically, before merging such prs, I'd like to see some kind of proof that generated extrinsic actually works. This could be...
> 
> Good suggestion, thank you.
> 
> For these PRs, I will not have the means to test the transactions (no funds to shuttle across the chains repeatedly). I'm not interested in gaining access to Interlay's test accounts, either. It's more secure the fewer people have access, plus there will be extensive tests once this is added to the UI.
> 
> It's easy enough to provide (links to) the generated extrinsics, so providing them is a no-brainer. Would it make sense to search for and link similar xcm transactions (same token) to other chains for comparison if they exist?

It would help but that's not a guarantee it'll work due to differences in e.g. existential deposit & fees. 

I think it's better to test as early as possible, rather than only after it gets added to UI. I get your point re: security, but if it's just a test account with < 100 usd worth of assets it's not much of a risk. I'd say it's worth it, especially since it also saves money in dev time if we catch problems as early as possible (no context switching, no coordinating of different people, no downstream redeployments, etc)

**Comment by bvotteler** [comment]
**Local testing (chopsticks) - transferring KBTC**
Succeeded with Alice account.

Kintsugi -> Karura
Kintsugi:
Screen Shot 2023-02-22 at 16 04 12
Karura:
Screen Shot 2023-02-22 at 16 04 25

Karura -> Kintsugi
Karura:
Screen Shot 2023-02-22 at 16 07 23
Kintsugi:
Screen Shot 2023-02-22 at 16 07 33

File: src/adapters/interlay.ts (hunk 1 above)
Comment (bvotteler) [reply src/adapters/interlay.ts:68]: Confirmed amount is safe in tests. Fees were right around the 123 mark.

File: src/adapters/interlay.spec.ts [hunk 2]
```diff
@@ -19,6 +91,31 @@ describe.skip("interlay-adapter should work", () => {
...
+    });
+
+    expect(
+      bridge.router.getDestinationChains({
+        from: chains.kintsugi,
+        token: "KBTC",
+      }).length
+    ).toEqual(1);
+
+    await runMyTestSuite(testAccount, bridge, "kintsugi", "heiko", "KBTC");
```
Comment (sander2) [inline src/adapters/interlay.spec.ts:116]: I think this only tests one direction, right? Shouldn't we do both?

File: src/adapters/interlay.ts (hunk 1 above)
Comment (sander2) [reply src/adapters/interlay.ts:68]: Wait, given the calculation, I assumed this was for the heiko -> kintsugi direction. But it seems from the naming it's the other direction? Then we can't apply this calculation

File: src/adapters/interlay.spec.ts (hunk 2 above)
Comment (bvotteler) [reply src/adapters/interlay.spec.ts:116]: > I think this only tests one direction, right? Shouldn't we do both?

Correct. This test file is for testing outgoing XCM from interlay/kintsugi.
We could add a parallel/heiko spec file for the other direction, too. 

I personally am happy enough if I can see the return direction's extrinsics generated in `bridge.spec.ts` which are then used for tests on chopsticks.
Having said that, I can see value in adding parallel specs, too.

File: src/adapters/interlay.ts (hunk 1 above)
Comment (bvotteler) [reply src/adapters/interlay.ts:68]: Interesting. During chopsticks testing, I landed pretty much on the mark just around the 123 mark. The reverse direction (heiko -> kintsugi) was considerably less.

File: src/adapters/interlay.spec.ts (hunk 2 above)
Comment (bvotteler) [reply src/adapters/interlay.spec.ts:93]: Added the reverse direction in the latest [commit](https://github.com/interlay/bridge/pull/50/commits/fb918ff3f6884703fd25112eecf72a1a21c41758)

File: src/adapters/interlay.ts (hunk 1 above)
Comment (bvotteler) [reply src/adapters/interlay.ts:63]: Submitted [this commit](https://github.com/interlay/bridge/pull/50/commits/119aad6551ce5e4f4c670c6f0b307f74900ebd9f) to address the fee estimates.

**Review by sander2 — APPROVED** [review]

**Pull request merged**